src/components/InputForm.js

```diff
@@ -1,22 +1,44 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 
 class InputForm extends React.Component {
 
   handleSubmit(e) {
     e.preventDefault();
-    const location = this.refs.location.value;
-    this.props.fetchLocation(location);
-    this.refs.locationForm.reset();
+    const location = this.props.inputText;
+    if (location) {
+      this.props.fetchLocation(location);
+      this.props.changeInput('');
+    } else {
+      this.props.changeStatus('Input cannot be empty', 'red');
+    }
+  }
+
+  handleChange(e) {
+    this.props.changeInput(e.target.value);
```

Comment:
are you using auto complete? I don't think it makes sense to dispatch an action on every keystroke... you can store temporary values like this in the component's state.